app/code/Magento/CatalogSearch/Ui/DataProvider/Product/AddFulltextFilterToCollection.php

```diff
@@ -37,7 +37,7 @@ public function __construct(SearchCollection $searchCollection)
     public function addFilter(Collection $collection, $field, $condition = null)
     {
         /** @var $collection \Magento\Catalog\Model\ResourceModel\Product\Collection */
-        if (isset($condition['fulltext']) && !empty($condition['fulltext'])) {
+        if (isset($condition['fulltext']) && 0 !== strlen($condition['fulltext'])) {
```

Comment:
Hi, @jeysmook, thanks for collaboration. Please change the condition to `(string)$condition['fulltext'] !== ''`.